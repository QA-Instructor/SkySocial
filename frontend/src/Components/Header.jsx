import React from 'react'
import { Link,Outlet } from "react-router-dom";
import Nav from 'react-bootstrap/Nav';
import Navbar from 'react-bootstrap/Navbar';
import "bootstrap/dist/css/bootstrap.min.css"
import "../resources/header.css"


export const Header = () => {
  return (
        <>
        <header>
            <div className='header-logo-section'>
                <img alt='Sky Social'/>
            </div>

            <Navbar className='navbar'>
                <Nav className='nav'>
                    <ul className='nav-list-left'>
                        <li className='navbar-item'id='home'>
                            <Link className='nav-link' to='/'>Home</Link>
                        </li>
                        <li className='navbar-item' id='createEvent'>
                            <Link className='nav-link' to='/CreateEvent'>Create Event</Link>
                        </li>
                        <li className='navbar-item' id='myEvents'>
                            <Link className='nav-link' to='/MyEvents'>My Events</Link>
                        </li>
                        
                    </ul>
                    <ul className='nav-list-right'>
                    <li className='navbar-item' id='search'>
                            <Link className='nav-link' to='/Search'>Search</Link>
                        </li>
                        <li className='navbar-item' id='profile'>
                            <Link className='nav-link' to='/Profile'>Profile</Link>
                        </li>
                        <li className='navbar-item' id='login'>
                            <Link className='nav-link' to='/Login'>Logout</Link>
                        </li>
                    </ul>
                </Nav>
            </Navbar>
        </header>
        <Outlet/>
        </>
  )
}
export default Header;
