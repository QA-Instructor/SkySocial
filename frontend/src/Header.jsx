import React from 'react'
import { Link } from "react-router-dom";
import Nav from 'react-bootstrap/Nav';
import Navbar from 'react-bootstrap/Navbar';
import "bootstrap/dist/css/bootstrap.min.css"
import "./header.css"
export const Header = () => {
  return (
    <header>
        <div className='header-logo-section'>
            <img alt='Sky Social'/>
        </div>

        <Navbar className='navbar'>
            <Nav className='nav'>
                <ul className='nav-list-left'>
                    <li className='navbar-item'>
                        <Link className='nav-link' to='/home'>Home</Link>
                    </li>
                    <li className='navbar-item'>
                        <Link className='nav-link' to='/createEvent'>Create Event</Link>
                    </li>
                    <li className='navbar-item'>
                        <Link className='nav-link' to='/myEvents'>My Events</Link>
                    </li>
                    
                </ul>
                <ul className='nav-list-right'>
                <li className='navbar-item'>
                        <Link className='nav-link' to='/search'>Search</Link>
                    </li>
                    <li className='navbar-item'>
                        <Link className='nav-link' to='/profile'>Profile</Link>
                    </li>
                    <li className='navbar-item'>
                        <Link className='nav-link' to='/login'>Logout</Link>
                    </li>
                </ul>
            </Nav>
        </Navbar>
    </header>
  )
}
export default Header;
