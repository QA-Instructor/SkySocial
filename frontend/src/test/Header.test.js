import React from 'react';
import { create } from 'react-test-renderer';
import Header from '../Components/Header';

describe('Header Snapshot tests',()=>{
    test(`Header matches snapshot`, () => {
        const header = create(<Header />);
        expect(header.toJSON()).toMatchSnapshot();
    });
    test(`Searchbar appears`, () => {
        const searchBar = create(<SearchBar />);
        expect(searchBar.toJSON()).toMatchSnapshot();
    });    
});

describe("Content appears as expected",()=>{
    test(`logo appears`, () => {
        const testRenderer = create(<Header />);
        const testInstance = testRenderer.root;
        const logo = testInstance.findByType(`img`);
        expect(logo.props.name).toContain("logoImage");
        expect(logo.props.src).toContain("logo.jpeg");
    });
    test(`home button appears`, () => {
        const testRenderer = create(<Header />);
        const testInstance = testRenderer.root;
        const button = testInstance.getByText(/home/i);
        expect(button).toBeInTheDocument();
    });

})
describe('Navbar functionalities tests',()=>{
    test(`Can navigate to profile page from home page`, () => {
        const testRenderer = create(<Header />);
        const testInstance = testRenderer.root;
        const homeButton = testInstance.getByText(/home/i);
        fireEvent.click(homeButton);
        expect(global.window.location.pathname).toContain('/home');
        const profileButton = testInstance.getByText(/profile/i);
        fireEvent.click(profileButton);
        expect(global.window.location.pathname).toContain('/profile');
    }); 
    test(`Can navigate to create event page from my events page`, () => {
        const testRenderer = create(<Header />);
        const testInstance = testRenderer.root;
        const myEventsButton = testInstance.getByText(/my events/i);
        fireEvent.click(myEventsButton);
        expect(global.window.location.pathname).toContain('/myEvent');
        const createEventButton = testInstance.getByText(/create event/i);
        fireEvent.click(createEventButton);
        expect(global.window.location.pathname).toContain('/createEvent');
    });
    test(`Can logout`, () => {
        const testRenderer = create(<Header />);
        const testInstance = testRenderer.root;
        const logoutButton = testInstance.getByText(/logout/i);
        fireEvent.click(logoutButton);
        expect(global.window.location.pathname).toContain('/login');
    });  
});


