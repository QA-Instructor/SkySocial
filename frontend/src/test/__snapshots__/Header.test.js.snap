// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Snapshot tests Header matches snapshot 1`] = `
<header>
  <div
    className="header-logo-section"
  >
    <img
      alt="Sky Social"
    />
  </div>
  <nav
    className="navbar navbar navbar-expand navbar-light"
  >
    <div
      className="nav navbar-nav"
      onKeyDown={[Function]}
    >
      <ul
        className="nav-list-left"
      >
        <li
          className="navbar-item"
          id="home"
        >
          <a
            className="nav-link"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="navbar-item"
          id="createEvent"
        >
          <a
            className="nav-link"
            href="/CreateEvent"
            onClick={[Function]}
          >
            Create Event
          </a>
        </li>
        <li
          className="navbar-item"
          id="myEvents"
        >
          <a
            className="nav-link"
            href="/MyEvents"
            onClick={[Function]}
          >
            My Events
          </a>
        </li>
      </ul>
      <ul
        className="nav-list-right"
      >
        <li
          className="navbar-item"
          id="search"
        >
          <a
            className="nav-link"
            href="/Search"
            onClick={[Function]}
          >
            Search
          </a>
        </li>
        <li
          className="navbar-item"
          id="profile"
        >
          <a
            className="nav-link"
            href="/Profile"
            onClick={[Function]}
          >
            Profile
          </a>
        </li>
        <li
          className="navbar-item"
          id="login"
        >
          <a
            className="nav-link"
            href="/Login"
            onClick={[Function]}
          >
            Logout
          </a>
        </li>
      </ul>
    </div>
  </nav>
</header>
`;
